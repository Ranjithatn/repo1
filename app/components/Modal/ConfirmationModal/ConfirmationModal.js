import React, { Fragment } from "react";
import { connect } from "react-redux";
import { Link, withRouter } from "react-router-dom";
import Modal from "../Modal";
import Title from "../../Title/Title";
import Button from "../../Button/Button";
import Switch from "../../Switch/Switch";
import { requestHideModal } from "../../../actions/modal";

import "./ConfirmationModal.scss";
import { requestStopLatentScan } from "../../../actions/latent";
export const ConfirmationModal = ({
  requestHideModal,
  formatMessage,
  action,
  message,
  history,
  requestStopLatentScan
}) => {

  return (
    <Modal
      title={formatMessage({ id: "confirm" })}
      className="newAction-modal"
      requestHideModal={requestHideModal}
      content={formatMessage({id:message})}
      buttons={
        <Fragment>
          <Button
            id="newAction-btn"
            className="is-primary"
            text={formatMessage({ id: "cancel" })}
            onClick={requestHideModal}
          />
          <Button
            id="newAction-btn"
            className="is-primary"
            text={formatMessage({ id: "Yes" })}
            onClick={() => { action();requestHideModal();requestStopLatentScan() } }
          />
        </Fragment>
      }
    />
  );
};

const mapState = state => ({});
export default connect(mapState, {
  requestHideModal,
  requestStopLatentScan
})(withRouter(ConfirmationModal));
