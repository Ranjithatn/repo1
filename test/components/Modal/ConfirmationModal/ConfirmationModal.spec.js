import React from "react";
import toJson from "enzyme-to-json";
import { ConfirmationModal } from "../../../../app/components/Modal/ConfirmationModal/ConfirmationModal";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });

const props = {
  formatMessage: data => data.id
}

describe('<ConfirmationModal />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <ConfirmationModal {...props}/>
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
