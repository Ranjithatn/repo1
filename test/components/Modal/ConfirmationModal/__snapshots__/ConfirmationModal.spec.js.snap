// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationModal /> should render without crashing 1`] = `
<Component
  formatMessage={[Function]}
>
  <Modal
    buttons={
      <UNDEFINED>
        <Button
          className="is-primary"
          id="newAction-btn"
          onClick={undefined}
          text="cancel"
        />
        <Button
          className="is-primary"
          id="newAction-btn"
          onClick={[Function]}
          text="Yes"
        />
      </UNDEFINED>
    }
    className="newAction-modal"
    title="confirm"
  >
    <div
      className="modal is-active newAction-modal"
      style={
        Object {
          "zIndex": "50",
        }
      }
    >
      <div
        className="modal-background"
      />
      <div
        className="modal-card"
      >
        <header
          className="modal-card-head"
        >
          <p
            className="modal-card-title"
          >
            confirm
          </p>
          <button
            aria-label="close"
            className="delete is-large"
            name="btn-modal-close"
          />
        </header>
        <section
          className="modal-card-body"
        />
        <footer
          className="modal-card-foot"
        >
          <Button
            className="is-primary"
            id="newAction-btn"
            text="cancel"
          >
            <button
              className="button is-primary"
              disabled={false}
              id="newAction-btn"
              style={Object {}}
              type="button"
            >
              <span>
                cancel
              </span>
            </button>
          </Button>
          <Button
            className="is-primary"
            id="newAction-btn"
            onClick={[Function]}
            text="Yes"
          >
            <button
              className="button is-primary"
              disabled={false}
              id="newAction-btn"
              onClick={[Function]}
              style={Object {}}
              type="button"
            >
              <span>
                Yes
              </span>
            </button>
          </Button>
        </footer>
      </div>
    </div>
  </Modal>
</Component>
`;
